tests/build_parents_file.sh

```diff
@@ -0,0 +1,251 @@
+#!/bin/bash
+
+POSITIONAL_ARGS=()
+VERBOSE="false"
+
+while [[ $# -gt 0 ]]; do
+  case $1 in
+    -v|--verbose)
+      VERBOSE="true"
+      shift # past argument
+      ;;
+    -*|--*)
+      echo "Unknown option $1"
+      exit 1
+      ;;
+    *)
+      POSITIONAL_ARGS+=("$1") # save positional arg
+      shift # past argument
+      ;;
+  esac
+done
+
+set -- "${POSITIONAL_ARGS[@]}" # restore positional parameters
+
+# default samples file path
+samples_file=$(pwd)/extraDevfileEntries.yaml
+# Cached remote samples directory
+samples_dir=$(pwd)/samples/.cache
+# default stacks directory
+stacks_dir=${STACKS:-"$(pwd)/stacks"}
+parents_file=$(pwd)/parents.yaml
+# Base directory path
+base_path=$(realpath $(dirname $(dirname $0)))
+
+# YAML query cmd path
+YQ_PATH=${YQ_PATH:-yq}
+
+# Read samples file as first argument
+# if unset use default samples file path
+if [ ! -z "${1}" ]; then
+    samples_file=${1}
+fi
+
+get_parent_version() {
+    devfile=$1
+    name=$2
+    version=$($YQ_PATH eval .parent.version ${devfile})
+
+    if [ "${version}" == "null" ] && [ -f "${stacks_dir}/${name}/stack.yaml" ]; then
+        version=$($YQ_PATH eval '.versions | filter(.default) | .[0].version' ${stacks_dir}/${name}/stack.yaml)
+    fi
+
+    echo ${version}
+}
+
+# Get parent index if exists, else returns -1
+parent_index() {
+    name=$1
+    version=$2
+
+    if [ -z "${version}" ]; then
+        result=$($YQ_PATH eval ".parents | to_entries | filter(.value.name == \"${name}\") | .[0].key" ${parents_file})
+    else
+        result=$($YQ_PATH eval ".parents | to_entries | filter(.value.name == \"${name}\" and .value.version == \"${version}\") | .[0].key" ${parents_file})
+    fi
+
+    if [ "${result}" == "null" ]; then
+        echo "-1"
+    else
+        echo ${result}
+    fi
+}
+
+# Get child index if exists, else returns -1
+child_index() {
+    parent_idx=$1
+    name=$2
+    version=$3
+
+    if [ -z "${version}" ]; then
+        result=$($YQ_PATH eval ".parents.[${parent_idx}].children | to_entries | filter(.value.name == \"${name}\") | .[0].key" ${parents_file})
+    else
+        result=$($YQ_PATH eval ".parents.[${parent_idx}].children | to_entries | filter(.value.name == \"${name}\" and .value.version == \"${version}\") | .[0].key" ${parents_file})
+    fi
+
+    if [ "${result}" == "null" ]; then
+        echo "-1"
+    else
+        echo ${result}
+    fi
+}
+
+# Builds sample parents
+build_parents() {
```

Comment:
Similar to `build_children` does it makes sens to rename this to `build_parent`?